column/api/manager/run_manager.py

```diff
@@ -0,0 +1,79 @@
+import copy
+import logging
+import threading
+
+import column
+from column.api import backend
+from column.api.objects import State
+from column.plugins.callback import progress
+
+
+LOG = logging.getLogger(__name__)
+
+
+class RunManager(object):
+    """Column api manager class
+
+    The Manager layer is to support additional logic which is needed
+    to run ansible playbook.
+
+    Attributes:
+        column_manager (obj): column library interface for running ansible
+            playbook
+    """
+
+    def __init__(self):
+        self.column_manager = column.APIRunner()
+        self.backend_store = backend.get_store()
+
+    def _build_opts(self, opts):
+        column_opts = copy.deepcopy(opts)
+        # state attribute is not useful to column runner
+        if 'state' in column_opts:
+            del column_opts['state']
+        # state attribute is not useful to column runner
+        if 'id' in column_opts:
+            del column_opts['id']
+        return column_opts
+
+    def _parse_result(self, run_id, result):
+        unreachable_hosts = result['unreachable_hosts']
+        failed_hosts = result['failed_hosts']
+        run = self.backend_store.get_run(run_id)
+        if unreachable_hosts:
+            error_msg = ("{} failed because the following nodes were "
+                         "unreachable: {}.\n".format(run['playbook_path'],
+                unreachable_hosts))
+            run['state'] = State.ERROR
+            run['message'] = error_msg
+            self.backend_store.update_run(run_id, run)
+            return
```

Comment:
This is a strange series of if-blocks.  Just do if, elif, else. No need to put empty return statements.